sw/device/lib/dif/dif_aon_timer.h

```diff
@@ -0,0 +1,272 @@
+ * `dif_aon_timer_wakeup_restart`.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_stop(const dif_aon_timer_t *handle);
+
+/** Restarts Always-On Timer (wake-up timer).
+ *
+ * Clears the counter and restarts the timer using the existing configuration.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_restart(
+    const dif_aon_timer_t *handle);
+
+/** Retrieves Always-On Timer (wake-up timer) tick count.
+ *
+ * @param handle An Always-On Timer handle.
+ * @param[out] count Current timer tick count.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_wakeup_get_count(
+    const dif_aon_timer_t *handle, uint32_t *count);
+
+/** Starts Always-On Timer (watchdog timer).
+ *
+ * @param handle A Always-On Timer handle.
+ * @param bark_threshold "Bark" threshold in ticks.
+ * @param bite_threshold "Bite" threshold in ticks.
+ * @param pause_in_sleep Watchdog is paused when device is in one of low power
+ *                       modes.
+ * @param lock Lock access to watchdog configuration registers.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_start(
+    const dif_aon_timer_t *handle, uint32_t bark_threshold,
+    uint32_t bite_threshold, bool pause_in_sleep, bool lock);
+
+/** Stops Always-On Timer (watchdog timer).
+ *
+ * Stops the timer. Configuration is not touched, and can be restarted via
+ * `dif_aon_timer_watchdog_restart`.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_stop(
+    const dif_aon_timer_t *handle);
+
+/** Restarts Always-On Timer (watchdog timer).
+ *
+ * Clears the counter and restarts the timer using the existing configuration.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_restart(
+    const dif_aon_timer_t *handle);
+
+/** Retrieves Always-On Timer (watchdog timer) tick count.
+ *
+ * @param handle An Always-On Timer handle.
+ * @param[out] count Current timer tick count.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_watchdog_get_count(
+    const dif_aon_timer_t *handle, uint32_t *count);
+
+/** Clears Always-On Timer (watchdog timer).
+ *
+ * This function must be called periodically to satisfy "Bite" and "Bark"
+ * thresholds.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_pet_watchdog(
+    const dif_aon_timer_t *handle);
```

Comment:
This is the same as the corresponding `restart` function, but doesn't touch the watchdog control register - hence will always succeed, even when watchdog is locked.